doc/languages.rst

```diff
@@ -846,6 +846,14 @@ to view the docstring of the syntax checker.  Likewise, you may use
          The type of the crate to check, as string for the ``--crate-type``
          option.
 
+      .. option:: flycheck-rust-binary-name
+
+         The name of the binary to pass to ``cargo rustc --bin``, as a string.
+
+         `rust-cargo` needs this option to correctly invoke cargo when
+         `flycheck-rust-crate-type` is ``bin`` in the presence of multiple
+         binary targets.
```

Comment:
Replace this entire paragraph with `Only required when flycheck-rust-crate-type is bin`.   This page only provides an overview over all options, we shouldn't go into too much detail here.  Details go into the docstrings 😃